source/common/utility.h

```diff
@@ -54,6 +55,15 @@ class Utility {
    * @param argv forwarded argv argument of the main entry point.
    */
   static void parseCommand(TCLAP::CmdLine& cmd, const int argc, const char* const* argv);
+
+  /**
+   * @param host_port host:port as a string, where host can be IPv4, [IPv6], or a DNS
+   * name.
+   * @param host string* to receive the host if the parse succeeds
+   * @param port int* to receive the port if the parse succeeds
+   * @return bool true if the input could be parsed as host:port
+   */
+  static bool parseHostPort(const std::string& host_port, std::string* host, int* port);
```

Comment:
Optional -- One potential alternative internal implementation is to construct a mock temp `tcp://host:port` uri based on the host/port, and rely on `Envoy::Network::Utility::portFromTcpUrl()` / `Envoy::Network::Utility::hostFromTcpUrl()`
(I think Envoy also does that in some places): https://github.com/envoyproxy/envoy/blob/21aa1a72ed976288ccd5b1612cc9d20f72e5a25d/source/common/network/utility.h#L111

Reply:
I tried those out, but unfortunately they don't seem to work with the colons in IPv6 addresses -- malformed url: tcp://[::1]:5" thrown in the test body.

I don't see any existing utility that supports a mixture of DNS, IPv4, IPv6, other than the UriImpl which comes into play later.

Reply:
Ah, I see, LGTM as is then